packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx

```diff
@@ -244,6 +251,19 @@ export default function HierarchyExplorer({ appId, className }: HierarchyExplore
     [dom, domApi],
   );
 
+  const handlePageSettingsNode = React.useCallback(
+    (nodeId: NodeId) => {
+      if (nodeId) {
+        const node = appDom.getNode(dom, nodeId);
+        if (appDom.isPage(node)) {
+          domApi.setView({ kind: 'page', nodeId: node.id });
```

Comment:
This can be done in a single action with domApi.update(dom => dom, { view: { kind: 'page', nodeId: node.id }, selectedNodeId: null }). I have a PR open to make it possible to update just the view and the selected node without "updating" the app DOM, that should make this less weird.

Reply:
@apedroferreira Updated in https://github.com/mui/mui-toolpad/pull/1672/commits/a87f8b75e4caff4fe38acbedd884eed0ba8d304f!